internal/core/model/adapters.rs

```diff
@@ -1065,6 +1067,11 @@ where
         container.wrapped_model.model_tracker().attach_peer(container.as_ref().model_peer());
         Self(container)
     }
+
+    /// Returns a reference to the inner model
+    pub fn source_model(&self) -> Pin<&M> {
```

Comment:
Interesting, is there a particular reason for pinning here? If, like in your test, the source model is an `Rc<dyn Model>` (aka `ModelRc`) then wouldn't a `&M` be fine as well?

Reply:
We can't return `&M` as trying to do so would result in a compile error `return a reference to a temporary` (a.k.a the `Pin<>` from the call to `ModelChangeListenerContainer::get()`)

Reply:
Or maybe with some unpining ?

Reply:
I can unpin, and I should !

Reply:
Fixed